modules/gdscript/gd_parser.cpp

```diff
@@ -1515,9 +1515,17 @@ GDParser::Node *GDParser::_reduce_expression(Node *p_node, bool p_to_const) {
 						return p_node;
 					}
 
-					ConstantNode *cn = alloc_node<ConstantNode>();
-					cn->value = v;
-					return cn;
+					if (v.get_type() == Variant::ARRAY) {
```

Comment:
I'm not completely sure about this. If parsing a constant, you will probably get an actual array (the array node will not work).
Also there may be a chance of the array having something in such case? In any case, I would put this code in the variable parser, not sure if here.

Reply:
I was hesitant too that this was the right location.  It works, but it may not work in cases I cannot begin to imagine.  I tried to no avail in parse_expression, but I'll give it another go.